linkerd/meshtls/src/server.rs

```diff
@@ -74,14 +93,18 @@ impl Server {
 
 impl<I> Service<I> for Server
 where
-    I: io::AsyncRead + io::AsyncWrite + Send + Sync + Unpin + 'static,
+    I: io::AsyncRead + io::AsyncWrite + Send + Unpin + 'static,
 {
     type Response = (ServerTls, ServerIo<I>);
     type Error = io::Error;
     type Future = TerminateFuture<I>;
 
+    #[inline]
     fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
         match self {
+            #[cfg(feature = "boring")]
+            Self::Boring(svc) => <boring::Server as Service<I>>::poll_ready(svc, cx),
```

Comment:
looking at this code it occurs to me that it's _probably_ possible to have a macro that, given a list of feature names and type names, generates all of these match arms automagically...but this is probably not worth bothering with unless we end up supporting a larger number of TLS implementations...